src/device/wg_controller.rs

```diff
@@ -200,12 +199,60 @@ async fn consolidate_wg_peers<
     for key in delete_keys {
         telio_log_info!("Removing peer: {:?}", actual_peers.get(key));
         wireguard_interface.del_peer(*key).await?;
+
+        // Remove from session_keeper
+        if let Some(sk) = session_keeper {
+            sk.remove_node(key).await?;
```

Comment:
I am feeling cautious about this one - Libtelio currently assumes that SessionKeeper is only used by direct peers and `device.rs` has: 
```                             
if !was_proxying && is_proxying {
...
2293: direct_entities.session_keeper.remove_node(&public_key).await?;
```

So there's a risk that actually both are happening one after another(I am not sure). One would then short-circuit another and cause a failure.

Of course that might not be the case and I am just paranoid :D